feat: implement resizable panels with drag support

iRajatDas opened the pull request:
## Summary
- Implemented resizable panels.
- Users can resize panels by dragging the middle divider.
- Ensured responsiveness and tested on multiple screen sizes.

## How to Test
1. Run the project.
2. Try resizing the panels by dragging the divider.
3. Check if the layout adjusts correctly without breaking.

## Related Issues
Closes #31

giacomocerquone reviewed (changes requested):
Also, you reformatted the files without using biome.

please run `npm run biome:fix` before committing (I think on this we should add a pre-commit hook honestly)

giacomocerquone commented on src/ui/Header/Header.module.css:
```diff
@@ -7,6 +7,9 @@
   font-size: var(--mantine-font-size-xl);
   font-weight: 700;
   margin: 0;
+  @media screen and (max-width: 768px) {
+    font-size: var(--mantine-font-size-sm);
+  }
```
nice touch with this one 👍

giacomocerquone commented on src/App.tsx:
```diff
@@ -86,100 +112,122 @@ const App = () => {
+              value={schema}
+            />
+          </ResizablePanel>
+          <ResizableHandle />
```
we probably want the handles (especially on mobile), so please add `withHandle`

giacomocerquone commented on src/ui/Resizable/resizable.tsx:
```diff
@@ -0,0 +1,61 @@
+import * as ResizablePrimitive from "react-resizable-panels";
+import classes from "./Resizable.module.css";
+
+const GripVertical = (props: React.ComponentProps<"svg">) => (
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    width={24}
+    height={24}
+    viewBox="0 0 24 24"
+    fill="none"
+    stroke="currentColor"
+    strokeWidth={2}
+    strokeLinecap="round"
+    strokeLinejoin="round"
+    className="lucide lucide-grip-vertical"
+    {...props}
+  >
+    <title>Grip Vertical Icon</title>
+    <circle cx={9} cy={12} r={1} />
+    <circle cx={9} cy={5} r={1} />
+    <circle cx={9} cy={19} r={1} />
+    <circle cx={15} cy={12} r={1} />
+    <circle cx={15} cy={5} r={1} />
+    <circle cx={15} cy={19} r={1} />
+  </svg>
+);
```
this is not needed. If you'd have inspected the package.json, you'd have noticed that this project uses react-icons.

So you can use:

import { RxDragHandleDots2 } from "react-icons/rx";
<RxDragHandleDots2 />

giacomocerquone commented on src/ui/Resizable/Resizable.module.css:
So, this project doesn't use tailwind and even though this css file doesn't require much from it (apart from some of its vars like `--tw-ring-offset-width`) it's definitely bloated css, or at least not coherent with the rest of the project, that I'd like to have removed.

Furthermore, please try to reuse existing vars from [mantine.dev](https://mantine.dev/) (check for them by inspecting the browser's style inspector)

iRajatDas replied:
Makes sense! I'll replace the Tailwind variables with their actual values. Thanks

marilari88 commented:
Hi @iRajatDas , any news for this PR?

iRajatDas commented:
> Hi @iRajatDas , any news for this PR?

I apologize for the long delay in addressing the PR feedback. I've finally made all the requested changes:

1. Removed pnpm-lock.yaml as requested and switched to npm
2. Cleaned up the CSS by replacing Tailwind variables with Mantine's design system
3. Added withHandle prop to the ResizablePanel component
4. Removed custom SVG in favor of react-icons (RxDragHandleDots2)
5. Ran biome formatter for consistent code style

All tests are passing and the changes have been kept minimal to address just the review feedback. Please let me know if there's anything else you'd like me to adjust.

Thanks

marilari88 commented on src/ui/Resizable/Resizable.module.css:
```diff
@@ -0,0 +1,75 @@
+/* 
+  These classes are generated by Tailwind CSS Playground.
+ */
```
Maybe we can remove this comment

marilari88 commented on src/ui/Resizable/Resizable.module.css:
```diff
@@ -0,0 +1,75 @@
+  }
+}
+
+@property --tw-content {
+  syntax: "*";
+  initial-value: "";
+  inherits: false;
+}
+
+@property --tw-translate-x {
+  syntax: "*";
+  inherits: false;
+  initial-value: 0;
+}
+
+@property --tw-translate-y {
+  syntax: "*";
+  inherits: false;
+  initial-value: 0;
+}
+
+@property --tw-border-style {
+  syntax: "*";
+  inherits: false;
+  initial-value: solid;
+}
+@property --tw-content {
+  syntax: "*";
+  initial-value: "";
+  inherits: false;
+}
```
What's the purpose of these properties?

marilari88 commented:
@iRajatDas You don't need to apologize. It's only right that we work on this project when we have the time and energy to do so. I left just few comments

marilari88 commented on package.json:
```diff
@@ -24,8 +24,9 @@
     "monaco-editor": "^0.52.2",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
-    "react-icons": "^5.4.0",
-    "typescript": "^5.7.2"
```
we need to solve the conflicts by restoring version of react-icons and typescript from `main`

iRajatDas commented on src/ui/Resizable/resizable.tsx:
```diff
@@ -0,0 +1,35 @@
+const ResizablePanelGroup = ({
+  className,
+  ...props
+}: React.ComponentProps<typeof ResizablePrimitive.PanelGroup>) => (
```
> Why don't you use directly `ResizablePrimitive.PanelGroupProps`?

It was from shadcn's component, so I kept it that way but converted the tailwindcss into vanilla.

marilari88 replied:
In the end, it doesn't make much of a difference. You can leave it as it is if you prefer
